x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_validator/job_validator.ts

```diff
@@ -37,7 +37,7 @@ export class JobValidator {
   private _jobCreator: JobCreatorType;
   private _validationSummary: ValidationSummary;
   private _lastJobConfig: string;
-  private _validateTimeout: NodeJS.Timeout;
+  private _validateTimeout: NodeJS.Timeout | null = null;
```

Comment:
This was already part of the existing code but I think we should try to get rid of `NodeJS.Timeout` for the client side code. An alternative could be to use `window.setTimeout` instead of `setTimeout` which returns just `number` as a type.

Reply:
good point! looking at other examples, it's also possible to use `ReturnType` to get the type of `setTimeout`. I think i'll use that method, rather than changing the call to `setTimeout`